app/src/main/java/org/simple/clinic/newentry/DateOfBirthFormatValidator.kt

```diff
@@ -2,24 +2,39 @@ package org.simple.clinic.newentry
 
 import org.simple.clinic.patient.PatientRepository
 import org.threeten.bp.LocalDate
+import org.threeten.bp.ZoneOffset
 import org.threeten.bp.format.DateTimeParseException
+import javax.inject.Inject
 
-object DateOfBirthFormatValidator {
+// TODO: Test that patient-entry-controller detects future-date-error and correctly updates the UI.
+
+class DateOfBirthFormatValidator @Inject constructor() {
 
   enum class Result {
     VALID,
-    INVALID
+    INVALID_PATTERN,
+    DATE_IS_IN_FUTURE,
   }
 
   fun validate(dateText: String): Result {
+    return validate(dateText, LocalDate.now(ZoneOffset.UTC))
+  }
+
+  fun validate(dateText: String, nowDate: LocalDate): Result {
```

Comment:
I added an overloaded function so that I can manually provide a value for `nowDate` in tests. Is there a better way to do this?

https://github.com/simpledotorg/simple-android/blob/3871d5ddbbb4c1bfdc0be9f287de63db4d427576/app/src/test/java/org/simple/clinic/newentry/DateOfBirthFormatValidatorTest.kt#L21

Reply:
Fixed